docs/Makefile

```diff
@@ -2,7 +2,7 @@
 #
 
 # You can set these variables from the command line.
-SPHINXOPTS    = -W --keep-going -n -jauto
+SPHINXOPTS    = --fail-on-warning -n -jauto
```

Comment:
This change is causing the HTML build to fail on my machine:

```sh
$ make html
sphinx-build -b html -d _build/doctrees  --fail-on-warning -n -jauto . _build/html
usage: sphinx-build [OPTIONS] SOURCEDIR OUTPUTDIR [FILENAMES...]
sphinx-build: error: unrecognized arguments: --fail-on-warning
make: *** [html] Error 2
```

Reply:
Looks like the long option was added in Sphinx 7.3, so this just needs a version bump in setup.py.